dashboard/test/lib/school_info_interstitial_helper_test.rb

```diff
@@ -87,4 +87,18 @@ class SchoolInfoInterstitialHelperTest < ActiveSupport::TestCase
     school_info.school_type = SchoolInfo::SCHOOL_TYPE_CHARTER
     refute SchoolInfoInterstitialHelper.complete? school_info
   end
+
+  test 'school info confirmation dialog is shown to a user when school info is nil' do
```

Comment:
This is a great start but can we add a few more cases? `show_school_info_confirmation_dialog` has a lot of logic in it and it would be great to see what we expect the output to be in a few "expected" cases.